editor/plugins/bone_map_editor_plugin.cpp

```diff
@@ -0,0 +1,549 @@
+		if (!sk) {
+			return;
+		}
+		skeleton = sk;
+	} else {
+		// Editor should not exist.
+		skeleton = nullptr;
+	}
+}
+
+void BoneMapEditor::_notification(int p_what) {
+	switch (p_what) {
+		case NOTIFICATION_ENTER_TREE: {
+			fetch_objects();
+			create_editors();
+		} break;
+		case NOTIFICATION_EXIT_TREE: {
+			remove_child(bone_mapper);
+			bone_mapper->queue_delete();
+		}
+	}
+}
+
+BoneMapEditor::BoneMapEditor(Ref<BoneMap> &p_bone_map) {
+	bone_map = p_bone_map;
+}
+
+BoneMapEditor::~BoneMapEditor() {
+}
+
+bool EditorInspectorPluginBoneMap::can_handle(Object *p_object) {
+	return Object::cast_to<BoneMap>(p_object) != nullptr;
+}
+
+void EditorInspectorPluginBoneMap::parse_begin(Object *p_object) {
+	BoneMap *bm = Object::cast_to<BoneMap>(p_object);
+	if (!bm) {
+		return;
+	}
+	Ref<BoneMap> r(bm);
+	editor = memnew(BoneMapEditor(r));
+	add_custom_control(editor);
+}
+
+BoneMapEditorPlugin::BoneMapEditorPlugin() {
+	// Register properties in editor settings.
+	EDITOR_DEF("editors/bone_mapper/handle_colors/set", Color(0.1, 0.6, 0.25));
+	EDITOR_DEF("editors/bone_mapper/handle_colors/error", Color(0.8, 0.2, 0.2));
+	EDITOR_DEF("editors/bone_mapper/handle_colors/unset", Color(0.3, 0.3, 0.3));
+
+	Ref<EditorInspectorPluginBoneMap> inspector_plugin;
+	inspector_plugin.instantiate();
+	add_inspector_plugin(inspector_plugin);
+
+	Ref<SkeletonPostImportPluginRenamer> post_import_plugin_renamer;
+	post_import_plugin_renamer.instantiate();
+	add_scene_post_import_plugin(post_import_plugin_renamer);
+}
+
+// Buil-in PostImportPlugins
+
+void SkeletonPostImportPluginRenamer::get_internal_import_options(InternalImportCategory p_category, List<ResourceImporter::ImportOption> *r_options) {
+	if (p_category == INTERNAL_IMPORT_CATEGORY_SKELETON_3D_NODE) {
+		r_options->push_back(ResourceImporter::ImportOption(PropertyInfo(Variant::BOOL, "retarget/bone_renamer/rename_bones"), true));
+	}
+}
+
+void SkeletonPostImportPluginRenamer::internal_process(InternalImportCategory p_category, Node *p_base_scene, Node *p_node, Ref<Resource> p_resource, const Dictionary &p_options) {
+	if (p_category == INTERNAL_IMPORT_CATEGORY_SKELETON_3D_NODE) {
+		// Prepare objects.
+		Object *map = p_options["retarget/bone_map"].get_validated_object();
+		if (!map || !bool(p_options["retarget/bone_renamer/rename_bones"])) {
+			return;
+		}
+		BoneMap *bone_map = Object::cast_to<BoneMap>(map);
+		Skeleton3D *skeleton = Object::cast_to<Skeleton3D>(p_node);
+
+		// Rename bones in Skeleton3D.
+		{
+			int len = skeleton->get_bone_count();
+			for (int i = 0; i < len; i++) {
+				StringName bn = bone_map->find_profile_bone_name(skeleton->get_bone_name(i));
+				if (bn) {
+					skeleton->set_bone_name(i, bn);
+				}
+			}
+		}
+
+		// Rename bones in Skin.
+		{
+			TypedArray<Node> nodes = p_base_scene->find_children("*", "ImporterMeshInstance3D");
+			while (nodes.size()) {
+				ImporterMeshInstance3D *mi = Object::cast_to<ImporterMeshInstance3D>(nodes.pop_front());
```

Comment:
Do not use pop_front() on an array.
Either reverse it up front, use pop_back() or perform a conventional for loop over the array elements (either for-each or index)

Reply:
Indeed, `pop_back()` is appropriate when considering the memory management process for arrays. Since the renaming process does not require parent-child ordering, `reverse()` should not be necessary.